src/complex/Common/BoundingBox.hpp

```diff
@@ -49,29 +47,45 @@ class BoundingBox
    * @param lowerLeft
    * @param upperRight
    */
-  BoundingBox(const PointType& lowerLeft, const PointType& upperRight)
-  : m_Bounds({lowerLeft[0], lowerLeft[1], lowerLeft[2], upperRight[0], upperRight[1], upperRight[2]})
+  BoundingBox(PointType lowerLeft, PointType upperRight)
+  : m_Lower(lowerLeft)
+  , m_Upper(upperRight)
   {
   }
 
   /**
-   * @brief Constructs a new BoundingBox defined by the array of position values.
+   * @brief Constructs a new BoundingBox3D defined by the array of position values.
    * The format is min X, min Y, min Z, max X, max Y, max Z.
    * @param arr
    */
-  BoundingBox(const std::array<T, 6>& arr)
-  : m_Bounds(arr)
+  explicit BoundingBox(const std::array<T, 6>& arr)
```

Comment:
```suggestion
  template <class = std::enable_if_t<PointType::Dimensions == 6>>
  explicit BoundingBox(const std::array<T, 6>& arr)
```
We should conditionally enable these constructors depending on the point type. This will require a static constexpr member in the `Vec` class.